### DISCUSSION TITLE
Optional tool usage confirmation

### DISCUSSION THREAD

**Discussion by Nighink** [modelcontextprotocol/modelcontextprotocol#71] [Ideas - General]
The spec says that clients should prompt for user confirmation on sensitive operations but I don't see a way to specify a way to label an operation as sensitive.

An example would be an SQL source where the client may Select to its heart's desire but should prompt for confirmation before initiating a Delete. This example would have two tools: select and delete. Delete would need a property with "sensitive". It would be good to then have an auxiliary function that translates the confirmation user prompt so it is "Do you want to delete all users?" instead of "Do you want to call the tool delete with 'DELETE FROM users'?"

Is there already a good possibility that I didn't spot or does someone have other ideas in this direction?

**Comment by localden** [comment]
We've limited Discussions in this repository to meeting notes and are closing this thread. To continue the conversation, pick a channel from [Contributor Communication](https://modelcontextprotocol.io/community/communication). Thanks for posting.
